### PR TITLE
task and media updates/requests updated to conform to new GQL schema

### PR DESCRIPTION
**Pull request by aaronshiel**

### CODE REVIEW

File: answer-transcode-mobile.py [hunk 1]
```diff
@@ -96,22 +93,20 @@ def handler(event, context):
     for record in event["Records"]:
         body = json.loads(str(record["body"]))
         request = json.loads(str(body["Message"]))["request"]
-        task_list = request["task_list"]
-        tasks = list(filter(lambda t: t["task_name"] == "transcoding-mobile", task_list))
-        if not tasks[0]:
+        task = request["transcodeMobileTask"] if "transcodeMobileTask" in request else None
```
Comment (mbsimonovic) [inline answer-transcode-mobile.py:96]: you've already modified the mentor-upload api for this new task format? https://github.com/mentorpal/mentor-upload/blob/56e6370ca83b37cee11a3e2667c24a4fa6f2eaa2/mentor_upload_api/src/mentor_upload_api/blueprints/upload/answer_queue.py#L129

Comment (mbsimonovic) [reply answer-transcode-mobile.py:96]: and you should also then update sample events https://github.com/mentorpal/mentor-upload-processor/blob/main/__events__/

File: answer-transcode-web.py [hunk 2]
```diff
@@ -93,22 +90,20 @@ def handler(event, context):
     for record in event["Records"]:
         body = json.loads(str(record["body"]))
         request = json.loads(str(body["Message"]))["request"]
-        task_list = request["task_list"]
-        tasks = list(filter(lambda t: t["task_name"] == "transcoding-web", task_list))
-        if not tasks:
+        task = request["transcodeWebTask"] if "transcodeWebTask" in request else None
```
Comment (mbsimonovic) [inline answer-transcode-web.py:93]: same here

File: answer-transcode-mobile.py (hunk 1 above)
Comment (aaronshiel) [reply answer-transcode-mobile.py:96]: yes, I have changes for all other services ready to deploy to keep everything consistent with one another

File: answer-transcribe-start.py [hunk 3]
```diff
@@ -119,22 +117,20 @@ def handler(event, context):
     for record in event["Records"]:
         body = json.loads(str(record["body"]))
         request = json.loads(str(body["Message"]))["request"]
-        task_list = request["task_list"]
-        tasks = list(filter(lambda t: t["task_name"] == "transcribing", task_list))
-        if not tasks:
+        task = request["transcribeTask"] if "transcribeTask" in request else None
```
Comment (mbsimonovic) [inline answer-transcribe-start.py:120]: same here

File: answer-transcode-mobile.py (hunk 1 above)
Comment (mbsimonovic) [reply answer-transcode-mobile.py:96]: this perhaps wasn't necessary, mentor-upload -> mentor-upload-processor sns/sqs message could use a list and then transform it here.

File: api.py [hunk 4]
```diff
@@ -82,7 +89,18 @@ def fetch_task_gql(mentor_id: str, question_id) -> GQLQueryBody:
     return {
         "query": """query UploadTask($mentorId: ID!, $questionId: ID!) {
             uploadTask(mentorId: $mentorId, questionId: $questionId){
-                taskList {
+                transcodeWebTask {
+                    task_name
```
Comment (mbsimonovic) [inline api.py]: you probably dont need this anymore, it was used with the list to identify task types

File: api.py [hunk 5]
```diff
@@ -94,6 +112,22 @@ def fetch_task_gql(mentor_id: str, question_id) -> GQLQueryBody:
         },
     }
 
+def convert_upload_task_gql(upload_task):
```
Comment (mbsimonovic) [inline api.py]: this probably isn't required, it's better to work with the map everywhere and avoid this map<->list conversion

File: answer-transcode-mobile.py (hunk 1 above)
Comment (aaronshiel) [reply answer-transcode-mobile.py:96]: ah you're right, well, no need to go back to the way it was before, right?

File: api.py (hunk 4 above)
Comment (aaronshiel) [reply api.py]: makes sense

File: api.py (hunk 5 above)
Comment (aaronshiel) [reply api.py]: okay, will do.

File: answer-transcode-mobile.py (hunk 1 above)
Comment (mbsimonovic) [reply answer-transcode-mobile.py:96]: nah leave it

File: api.py (hunk 4 above)
Comment (aaronshiel) [reply api.py]: [846abb0](https://github.com/mentorpal/mentor-upload-processor/pull/2/commits/846abb023e59f62ebf58924c9a089a13b536f0ae)

File: answer-transcode-mobile.py (hunk 1 above)
Comment (aaronshiel) [reply answer-transcode-mobile.py:96]: [846abb0](https://github.com/mentorpal/mentor-upload-processor/pull/2/commits/846abb023e59f62ebf58924c9a089a13b536f0ae) __events__ changes

File: api.py (hunk 5 above)
Comment (aaronshiel) [reply api.py]: [42855af](https://github.com/mentorpal/mentor-upload-processor/pull/2/commits/42855af608a5db04702f760347c173f9f29059f6)

File: answer-transcode-mobile.py [hunk 6]
```diff
@@ -38,10 +38,10 @@ def transcode_mobile(video_file, s3_path):
     )
 
 
-def process_task(request, task):
+def process_task(request):
     log.info("video to process %s", request["video"])
     stored_task = fetch_from_graphql(
-        request["mentor"], request["question"], task["task_id"]
+        request["mentor"], request["question"], "transcodeMobileTask"
```
Comment (mbsimonovic) [inline answer-transcode-mobile.py:44]: you sure about this? cant a mentor have multiple tasks for the same question, say submitting an answer in two tabs at the same time?

File: answer-transcribe-start.py (hunk 3 above)
Comment (mbsimonovic) [inline answer-transcribe-start.py:120]: you should probably update the transcribe sample event then

Comment (aaronshiel) [reply answer-transcribe-start.py:120]: answer-transcribe-start also uses answer-event.json.dist which contains the transcribeTask
answer-transcribe-collect uses transcribe-collect-event.json.dist which is already setup correctly

File: answer-transcode-mobile.py (hunk 6 above)
Comment (aaronshiel) [reply answer-transcode-mobile.py:44]: GraphQL has a unique constraint on the mentor and question fields for this document.

**Review by mbsimonovic — APPROVED** [review]

**Pull request merged**